modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Monoidal.kt

```diff
@@ -0,0 +1,109 @@
+package arrow.typeclasses
+
+import arrow.Kind
+import arrow.core.Option
+
+/**
+ * ank_macro_hierarchy(arrow.typeclasses.Monoidal)
+ *
+ * The [Monoidal] type class adds an identity element to the [Semigroupal] type class by defining the function [identity].
+ *
+ * [identity] returns a specific identity `Kind<F, A>` value for a given type [F] and [A].
+ *
+ * Like any other Monoid type class, [Monoidal] also complies with the left and right identity law:
```

Comment:
Same as in the laws, I thought that the product of `empty` and `a` is `a`, and not `empty`